README.md

```diff
@@ -54,6 +54,67 @@ Example configuration:
     ```
     See the [Kafka Connect transforms](https://kafka.apache.org/31/documentation.html#connect_transforms) documentation for more details.
 
+**Note:** For each message record that the sink connector writes to Cloudant the document ID from this priority order:
+
+1. From the value of the `cloudant_doc_id` header on the message.  The value passed to this header must be a string and the `header.converter=org.apache.kafka.connect.storage.StringConverter` config is required.  This will overwrite the `_id` field if it already exists.
+1. The value of the `_id` field in the JSON
+1. If no other non-null or non-empty value is available the document will be created with a new UUID.
+
+#### Single Message Transforms
+
+Single Message Transforms, or SMTs, can be used to customize fields or values of messages during data flow.  The examples below will explore modifying fields for messages flowing from the Kafka topic to a Cloudant database using the sink connector.
+
+1. If there's an existing field that will be used as the document ID then use the `ReplaceField` transform.  See the config example below that renames `name` to `_id` field:
+    ```
+    transforms=RenameField
+    transforms.RenameField.type=org.apache.kafka.connect.transforms.ReplaceField$Value 
+    transforms.RenameField.renames=name:_id
+    ```
+2. If you have `_id` fields and would prefer to have Cloudant generate a UUID for the document ID, use the `ReplaceField` transform to exclude the existing `_id` field:
+    ```
+    transforms=ReplaceField
+    transforms.ReplaceField.type=org.apache.kafka.connect.transforms.ReplaceField$Value 
+    transforms.ReplaceField.exclude=_id
+    ```
+3. If you have messages where the `_id` field is absent or `null` then Cloudant will generate
+a document ID. If you don't want this to happen then set an `_id` (see earlier examples).
+Alternatively filter out those documents. For example if you have messages where the `_id`
+field is `null` then you'll need to use a transform and predicate to filter out and remove this
+field:
+    ```
+    TODO 
+    ```
+
```

Comment:
I did some more digging and the current built-in SMT predicates and filter won't handle either a) dropping the field or b) filtering the document if `_id == null` .
At this point, I wouldn't have to delay this PR by trying to create a custom SMT.  What if we update this to:
```suggestion
3. If you have messages where the `_id` field is absent or `null` then Cloudant will generate
a document ID. If you don't want this to happen then set an `_id` (see earlier examples).
If you need to filter out those documents or drop `_id` fields when the value is `null` then you'll need to create a custom SMT.
```
I don't think this needs to be update a numbered point.  This could be a note at the end of this section.